packages/react-router/src/router.ts

```diff
@@ -70,7 +69,12 @@ import { NotFoundError, isNotFound } from './not-found'
 import { ResolveRelativePath, ToOptions } from './link'
 import { NoInfer } from '@tanstack/react-store'
 import warning from 'tiny-warning'
-import { DeferredPromiseState } from '.'
+import {
+  DeferredPromiseState,
+  NavigateOptions,
+  NotFoundRouteComponent,
+  notFound,
+} from '.'
```

Comment:
I would rather import from the files that export those types, e.g. `./defer` etc